modules/apps/frontend-js/frontend-js-tooltip-support-web/src/main/java/com/liferay/frontend/js/tooltip/support/web/internal/servlet/taglib/ClayTooltipBottomDynamicInclude.java

```diff
@@ -34,22 +45,43 @@ public void include(
 			HttpServletResponse httpServletResponse, String key)
 		throws IOException {
 
-		ScriptData scriptData = new ScriptData();
-
-		AbsolutePortalURLBuilder absolutePortalURLBuilder =
-			_absolutePortalURLBuilderFactory.getAbsolutePortalURLBuilder(
-				httpServletRequest);
-
-		scriptData.append(
-			null,
-			new JSFragment(
-				"main();",
-				Arrays.asList(
-					ESImportUtil.getESImport(
-						absolutePortalURLBuilder,
-						"{main} from frontend-js-tooltip-support-web"))));
-
-		scriptData.writeTo(httpServletResponse.getWriter());
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		Group group = themeDisplay.getScopeGroup();
+		Layout layout = themeDisplay.getLayout();
+
+		try {
+			JSTooltipSupportConfiguration jsTooltipSupportConfiguration =
+				_configurationProvider.getGroupConfiguration(
+					JSTooltipSupportConfiguration.class, group.getGroupId());
+
+			if (layout.isTypeControlPanel() ||
+				jsTooltipSupportConfiguration.enabled()) {
```

Comment:
pls return if conditions not match. This buys some space

Why check `layout.isTypeControlPanel()`? is it too risky not emmiting this in the CP pages? Or is it just we want setting to only act in site pages?

Reply:
I do this because CP isn't a performance concern and to keep consistency in our admin styling, we can keep it enabled.